### PR TITLE
WIP: Introduce socket APIs.

### PR DESCRIPTION
**Pull request by maxz-lab**
Add support for socket() and test cases in nplb.

socket functions:
  socket, connect, bind, listen, accept

Change-Id: Idcb6c113431ac8ff8bbd3fccaaf190a11f6cb8be

### CODE REVIEW

File: starboard/shared/modular/posix_socket_wrappers.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// limitations under the License.
+
+#ifndef STARBOARD_SHARED_MODULAR_POSIX_SOCKET_WRAPPERS_H_
+#define STARBOARD_SHARED_MODULAR_POSIX_SOCKET_WRAPPERS_H_
+
+#include <stdint.h>
+
+#include "starboard/configuration.h"
+#include "starboard/export.h"
+
```
Comment (y4vor) [inline starboard/shared/modular/posix_socket_wrappers.h:22]: Looks like you may not need wrappers. If that is the case please don't add the files.

File: starboard/nplb/socket_create_test.cc [hunk 2]
```diff
@@ -80,29 +82,7 @@ TEST_P(SbSocketCreateTest, ManyTcpAtOnce) {
   }
 }
 
-#if SB_HAS(IPV6)
```
Comment (y4vor) [inline starboard/nplb/socket_create_test.cc:83]: Why did you change the existing test?

Comment (maxz-lab) [reply starboard/nplb/socket_create_test.cc:83]: It is a mistake. The change will be removed.
